### PR TITLE
feat: passing data back and forth between Tauri client and Python

### PR DESCRIPTION
**Pull request by gjreda**
- https://github.com/refstudio/refstudio/issues/17

This updates the sidecar PoC. It does the following:
- Adds a python function that takes an input of text and counts the words in the text, outputting json to stdout
- Updates the AIView so that when text is highlights, the sidecar word count function is called, the response is read back into the tauri client and ultimately displayed in the view.

It's fun to see it working!

### CODE REVIEW

**Comment by cguedes** [comment]
@gjreda I noticed that after running `yarn python` files are created in`src-tauri/bin` and also the root file `main.spec`.
Should these be ignored (added to `.gitignore`) or tracked in git?

**Comment by gjreda** [comment]
@cguedes good catch! yes, we should ignore them from the repo since they will be specific to each of our local machines.

**Comment by hammer** [comment]
@cguedes perhaps you could be the reviewer here?

**Review by cguedes — APPROVED** [review]
@gjreda after merging main into this branch and installing all the dependencies again (`yarn install`, `poetry install` and `yarn python`) the app is now working fine 🎉 

image

@sergioramos I think you can merge this PoC into main (before #25 that is still in progress).

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
         "scope": [
           {
             "name": "bin/python/main",
-            "sidecar": true
+            "sidecar": true,
+            "args": true
```
[the remark this answers is not in the document]
Comment (gjreda) [reply src-tauri/tauri.conf.json:25]: @cguedes agree -- I think once we iron out the API between JS <> sidecar, we should restrict the arguments. I think that will come once we've determined which interactions we want to support for v1.

[the remark this answers is not in the document]
Comment (cguedes) [reply src-tauri/tauri.conf.json:25]: Great.

File: src/views/AIView.tsx [hunk 2]
```diff
@@ -27,9 +27,12 @@ export function AIView({ selection }: { selection: string | null }) {
 
 async function interactWithAi(selection: string) {
   try {
-    const aiReply = await invoke<string>('interact_with_ai', {
-      selection,
-    });
+    const command = Command.sidecar('bin/python/main', ['--text', `${selection}`]);
```
Comment (danvk) [inline src/views/AIView.tsx:30]: @cguedes @gjreda Is the idea here that `main` will eventually be the entrypoint for all Python commands? Or that we'll have separate Python scripts for all commands? I initially misread `bin/python/main` as being the `python` binary itself and started getting worried about security.

**Review by danvk — COMMENTED** [review]
I saw `Command.sidecar` in code, got curious and left some comments.

**Review by danvk — COMMENTED** [review]
I saw `Command.sidecar` in code, got curious and left some comments.

**Pull request merged**